File: MAStaff-Punishments/Punishments-API/src/main/java/es/angelillo15/mast/api/punishments/cmd/ban/BanCMD.java
package es.angelillo15.mast.api.punishments.cmd.ban;

import es.angelillo15.mast.api.cmd.Command;
import es.angelillo15.mast.api.cmd.CommandData;
import es.angelillo15.mast.api.cmd.sender.CommandSender;
import es.angelillo15.mast.api.punishments.IPunishPlayer;
import es.angelillo15.mast.api.punishments.PunishPlayersManager;
import es.angelillo15.mast.api.config.punishments.Messages;

@CommandData(
        name = "ban",
        description = "Ban a player",
        usage = "/ban <player> <reason>",
        permission = "mast.punishments.ban",
        aliases = {"b"}
)
public class BanCMD extends Command {
    @Override
    public void onCommand(CommandSender sender, String label, String[] args) {
        if (!(sender.hasPermission("mast.punishments"))) return;

        IPunishPlayer punishPlayer = PunishPlayersManager.getPlayer(sender.getUniqueId());

        if (args.length < 1) {
            sender.sendMessage(Messages.Commands.Ban.usage());
            return;
        }

        String target = args[0];

        StringBuilder reason = new StringBuilder();

        for (int i = 1; i < args.length; i++) {
            reason.append(args[i]).append(" ");
        }

        if (reason.toString().isEmpty()) {
            reason.append(Messages.Default.defaultBanReason());
        }

        if (target == null) {
            sender.sendMessage(Messages.Commands.playerNotFound(target));
            return;
        }

        new Thread(() -> {
            punishPlayer.ban(args[0], reason.toString());
            sender.sendMessage(Messages.Commands.Ban.success(target, reason.toString(), sender.getName()));
        }).start();
    }
}

File: MAStaff-Punishments/Punishments-API/src/main/java/es/angelillo15/mast/api/punishments/cmd/ban/BanCMD.kt
package es.angelillo15.mast.api.punishments.cmd.ban

import com.google.inject.Inject
import es.angelillo15.mast.api.IServerUtils
import es.angelillo15.mast.api.cmd.CommandData
import es.angelillo15.mast.api.cmd.sender.CommandSender
import es.angelillo15.mast.api.config.punishments.Messages
import es.angelillo15.mast.api.punishments.IPunishPlayer
import es.angelillo15.mast.api.punishments.cmd.PunishTargetReasonCommand
import es.angelillo15.mast.api.templates.managers.BanTemplatesManager

@CommandData(
    name = "ban",
    description = "Ban a player",
    usage = "/ban <player> <reason>",
    permission = "mast.punishments.ban",
    aliases = ["b"]
)
class BanCMD : PunishTargetReasonCommand(1, Messages.Default.defaultBanReason()) {
    @Inject
    private lateinit var serverUtils: IServerUtils
    @Inject
    private lateinit var banTemplatesManager: BanTemplatesManager

    override fun onCommand(sender: IPunishPlayer, target: String, label: String, args: Array<out String>, reason: String) {
        if (args.isEmpty()) {
            sender.sendMessage(Messages.Commands.Ban.usage())
            return
        }

        if (banTemplatesManager.getBanTemplate(reason) != null)
            sender.tryBanTemplate(target, reason)
        else
            sender.ban(target, reason)

        sender.sendMessage(Messages.Commands.Ban.success(target, reason, sender.name))
    }

    override fun onTabComplete(sender: CommandSender?, args: Array<String?>?): List<String?> {
        if (args!!.size == 1) {
            return serverUtils.getOnlinePlayersNames()
        }

        if (args.size == 2) {
            return banTemplatesManager.getBanTemplates().map { it.id }
        }

        return emptyList()
    }
}
